AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,87 @@
 //
 
 import UIKit
+import MobileCoreServices
+import AVKit
+import AVFoundation
 
-class DoubleVideoViewController: UIViewController, CellTitled {
+class DoubleVideoViewController: UIViewController, CellTitled, UIImagePickerControllerDelegate, UINavigationControllerDelegate {
     var titleForCell: String = "Double Video"
-
+    
+    var videoURL: URL?
+    var playerOne: AVPlayer?
+    var playerTwo: AVPlayer?
+    var playerLayerOne: AVPlayerLayer!
+    var playerLayerTwo: AVPlayerLayer!
+    var videoIsPlaying = false
+    
     @IBOutlet weak var videoContainerTop: UIView!
     @IBOutlet weak var videoContainerBottom: UIView!
     
     override func viewDidLoad() {
         super.viewDidLoad()
-
+        
         // don't f with this line
         self.edgesForExtendedLayout = UIRectEdge(rawValue: 0)
     }
-
+    
+    //MARK: - ImageProtocol Delegat Method
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+            self.videoURL = url
+        }
+        else {
+            print("Error getting url from picked asset")
+        }
+        dismiss(animated: true) {
+            if let url = self.videoURL {
+                if self.playerOne == nil {
+                    let playerItem = AVPlayerItem(url: url)
+                    self.playerOne = AVPlayer(playerItem: playerItem)
+                    let playerLayer = AVPlayerLayer(player: self.playerOne)
+                    playerLayer.frame = self.videoContainerTop.bounds
+                    self.videoContainerTop.layer.addSublayer(playerLayer)
+                    self.playerOne?.play()
+                    //self.videoURL = nil
+                }
+                else {
```

Comment:
This is why every video after the first falls through to here, even if there's one currently playing.